### ISSUE TITLE
Refactor Terraform to support AWS EKS

### ISSUE THREAD

**Issue by tgeoghegan** [abetterinternet/prio-server#655]
We want to be able to deploy `prio-server` clusters that run on AWS, using EKS, S3 and SNS/SQS. There should be a top-level parameter defined in an env's .tfvars file that governs whether the cluster is on GKE or EKS.

**Comment by tgeoghegan** [comment]
Progress update: I've got a whole bunch of Terraform changes to let us conditionally spin up either an EKS or GKE cluster based on the top-level `use_aws` variable. The difficulty is that Terraform is kinda hostile to refactoring, which means that applying these changes to existing environments (like prod or staging) would require a whole lot of perilous `terraform import` work to account for the changes to the layout of objects in Terraform (i.e., some resources that have moved from one module to another, or whose TF identifier now includes a `[0]` because we use a `count = some_condition ? 1 : 0` to conditionally create them). I'm thinking of just keeping the old and new TF modules in git so we don't have to block deployment of international on gracefully migrating the prod-us deployment.

Beyond that I have also updated deploy-tool so it can wrangle manifests in AWS S3 (that's #694). All those changes are in #699.

At this point, all the jobs and deployments in the EKS cluster launch. `workflow-manager` works normally and can scan buckets for tasks and schedule tasks (i.e., publish messages to SNS). However the `intake-batch` works fail when trying to authenticate to GCP in order to write objects to the peer validation bucket. So next up, I need to implement #681.
